store/store.go

```diff
@@ -1228,6 +1240,40 @@ var applyDelta = func(name string, deltaPath string, deltaInfo *snap.DeltaInfo)
 	return targetSnapPath, nil
 }
 
+// downloadAndApplyDelta downloads and then applies the delta to the current snap.
+func (s *Store) downloadAndApplyDelta(name, targetFn string, downloadInfo *snap.DownloadInfo, pbar progress.Meter, user *auth.UserState) (err error) {
+	deltaInfo := &downloadInfo.Deltas[0]
+	workingDir := filepath.Dir(targetFn)
+
+	deltaPath, err := s.downloadDelta(name, workingDir, downloadInfo, pbar, user)
```

Comment:
This doesn't seem aligned with the new convention, but rather just bringing back the original logic. Being unable to do that on time was one of the reasons why we dropped the patch right before the release.

Note that under the new convention, we tell where we want the filename to end up on, and the downloading function simply appends the ".partial" suffix.

To follow suit here, perhaps something like this could make sense:

1. Download delta from N to M onto: `<final filename>.xdelta-N-to-M.partial`
2. Apply delta and move to `<final filename>`

As some interesting consequences of that design, we know that everything under /snaps with the ".partial" suffix is optional/in-progress stuff that may be safely removed, and we also know that everything that is _not_ .partial is complete, and correct.

FWIW, I'm not a big fan of "targetFn" as a variable name as well.